src/header/map.rs

```diff
@@ -472,7 +472,9 @@ impl<T> HeaderMap<T> {
                 danger: Danger::Green,
             }
         } else {
-            let raw_cap = to_raw_capacity(capacity).next_power_of_two();
+            let raw_cap = to_raw_capacity(capacity)
+                .checked_next_power_of_two()
+                .expect("requested capacity overflows on checked_next_power_of_two");
```

Comment:
tiny nit, not a big deal: what do you think about changing this `expect` to a `match` and `panic!`, so that the panic message can include the requested capacity?
```suggestion
            let raw_cap =  match raw_capacity(capacity).checked_next_power_of_two() {
                Some(c) => c,
                None => panic!("requested capacity {capacity} too large: next power of two would overflow `usize`"),
            };
```

Reply:
reflected!